### PR TITLE
KAFKA-6962 added better docs to DESCRIBE_CONFIGS_REQUEST_RESOURCE_V0

### PR DESCRIPTION
**Pull request by kdrakon**

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/requests/DescribeConfigsRequest.java [hunk 1]
```diff
@@ -43,9 +44,9 @@
     private static final String CONFIG_NAMES_KEY_NAME = "config_names";
 
     private static final Schema DESCRIBE_CONFIGS_REQUEST_RESOURCE_V0 = new Schema(
-            new Field(RESOURCE_TYPE_KEY_NAME, INT8),
-            new Field(RESOURCE_NAME_KEY_NAME, STRING),
-            new Field(CONFIG_NAMES_KEY_NAME, ArrayOf.nullable(STRING)));
+            new Field(RESOURCE_TYPE_KEY_NAME, INT8, String.format("The resource type, which is one of %s", ResourceType.ID_NAME_LIST)),
+            new Field(RESOURCE_NAME_KEY_NAME, NULLABLE_STRING, "The resource name to query. If set to null, will retrieve all resources of type resource_type"),
```
Comment (omkreddy) [inline clients/src/main/java/org/apache/kafka/common/requests/DescribeConfigsRequest.java]: I think as per current implementation, resource name can not be null. We may get NullPointerExceptions along the code path.  Also server side implementation does not support the functionality explained in the description. 

KafkaAdminClient also does not support null values. https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/config/ConfigResource.java#L45

maybe you can write a unit test and check the behavior.

Comment (kdrakon) [reply clients/src/main/java/org/apache/kafka/common/requests/DescribeConfigsRequest.java]: You're right; I think I got confused with something else I was working on at the same time. Looking at this again, what I found using the binary protocol API was that the CONFIG_NAMES array was nullable. I'm pretty sure that's the only thing I wanted to indicate in docs was allowed to be null.

Comment (kdrakon) [reply clients/src/main/java/org/apache/kafka/common/requests/DescribeConfigsRequest.java]: This has been amended in the commit.

**Comment by omkreddy** [comment]
LGTM. maybe we can update AlterConfigsRequest also.

**Comment by kdrakon** [comment]
@omkreddy made those additions to AlterConfigsRequest as you suggested

**Comment by omkreddy** [comment]
LGTM. Let committers take a look.

**Pull request closed without merging**